src/lib/datepicker/datepicker.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  Component,
+  TemplateRef,
+  ViewChild,
+  ViewEncapsulation,
+  ChangeDetectionStrategy,
+  ViewContainerRef,
+  Optional,
+  ElementRef, OnDestroy
+} from '@angular/core';
+import {Overlay} from '../core/overlay/overlay';
+import {OverlayRef} from '../core/overlay/overlay-ref';
+import {TemplatePortal} from '../core/portal/portal';
+import {OverlayState} from '../core/overlay/overlay-state';
+import {Dir} from '../core/rtl/dir';
+import {MdError} from '../core/errors/error';
+import {MdDialog} from '../dialog/dialog';
+import {MdDialogRef} from '../dialog/dialog-ref';
+
+
+/** Component responsible for managing the datepicker popup/dialog. */
+@Component({
+  moduleId: module.id,
+  selector: 'md-datepicker',
+  templateUrl: 'datepicker.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdDatepicker implements OnDestroy {
+  /** Whether the calendar is in UI is in touch mode. */
```

Comment:
Typo: `is in UI is in touch mode` => `is in touch mode`

Also can you provide more info about what this mode means in the comment? I assume the interface is different for smaller screens?